RetailCoder.VBE/UI/SourceControl/SourceControlBindings.cs

```diff
@@ -34,15 +30,13 @@ public override void Load()
             Bind<IDeleteBranchView>().To<DeleteBranchForm>();
             Bind<IMergeView>().To<MergeForm>();
 
-            //presenters
-            Bind<IChangesPresenter>().To<ChangesPresenter>();
-            Bind<IBranchesPresenter>().To<BranchesPresenter>();
-            Bind<ISettingsPresenter>().To<SettingsPresenter>();
-            Bind<IUnsyncedCommitsPresenter>().To<UnsyncedCommitsPresenter>();
+            //factories 
+            // todo: check on note below
+            // ninject is complaining about also having a SourceControlProviderFactoryProxy and a FolderBrowserFactoryProxy
```

Comment:
The factory proxies are due to this method in the `RubberduckModule` class:

        // note convention: abstract factory interface names end with "Factory".
        private void ApplyAbstractFactoryConvention(IEnumerable<Assembly> assemblies)
        {
            _kernel.Bind(t => t.From(assemblies)
                .SelectAllInterfaces()
                .Where(type => type.Name.EndsWith("Factory"))
                .BindToFactory()
                .Configure(binding => binding.InSingletonScope()));
        }

Basically all abstract factories are automatically implemented by Ninject, which is great for the trivial implementations. Now, looking at that particular factory class, I *think* Ninject can handle it, but it would have to be tested; is there a binding for `ISourceControlProvider ~> GitProvider`?